v2/spiffeid/trustdomain_test.go

```diff
@@ -10,201 +11,129 @@ import (
-			expectedError: "spiffeid: invalid scheme",
-		},
-		{
-			name:          "missing_scheme",
-			input:         &url.URL{Host: "domain.test"},
-			expectedError: "spiffeid: invalid scheme",
-		},
-		{
-			name:          "has_port",
-			input:         parseURI(t, "spiffe://domain.test:80"),
-			expectedError: "spiffeid: port is not allowed",
-		},
-		{
-			name:          "has_colon",
-			input:         parseURI(t, "spiffe://domain.test:"),
-			expectedError: "spiffeid: colon is not allowed in trust domain",
-		},
+	parseURI := func(s string) *url.URL {
+		u, err := url.Parse(s)
+		require.NoError(t, err)
+		return u
 	}
-
-	for _, test := range tests {
-		test := test
-		t.Run(test.name, func(t *testing.T) {
-			td, err := spiffeid.TrustDomainFromURI(test.input)
-			if test.expectedError == "" {
-				assert.NoError(t, err)
-			} else {
-				assert.EqualError(t, err, test.expectedError)
-			}
-
-			assert.Equal(t, test.expectedTd, td.String())
-		})
+	assertOK := func(s string) {
+		u := parseURI(s)
+		td, err := spiffeid.TrustDomainFromURI(u)
+		assert.NoError(t, err)
+		assert.Equal(t, spiffeid.RequireTrustDomainFromString(u.Host), td)
+	}
+	assertFail := func(u *url.URL, expectErr string) {
+		_, err := spiffeid.TrustDomainFromURI(u)
+		assertErrorContains(t, err, expectErr)
 	}
-}
 
-func TestRequireTrustDomainFromURI(t *testing.T) {
-	// Just test the panic, the non panicing cases are already handled by TestTrustDomainFromURI.
-	assert.PanicsWithError(t, "spiffeid: colon is not allowed in trust domain", func() {
-		spiffeid.RequireTrustDomainFromURI(parseURI(t, "spiffe://domain.test:"))
-	})
+	assertOK("spiffe://trustdomain")
+	assertOK("spiffe://trustdomain/path")
+
+	assertFail(&url.URL{}, `invalid SPIFFE ID "": cannot be empty`)
+	assertFail(&url.URL{Scheme: "SPIFFE", Host: "trustdomain"}, `invalid SPIFFE ID "SPIFFE://trustdomain": scheme is missing or invalid`)
+	assertFail(parseURI("spiffe://trust$domain"), `invalid SPIFFE ID "spiffe://trust$domain": trust domain characters are limited to lowercase letters, numbers, dots, dashes, and underscores`)
+	assertFail(parseURI("spiffe://trustdomain/path$"), `invalid SPIFFE ID "spiffe://trustdomain/path$": path segment characters are limited to letters, numbers, dots, dashes, and underscores`)
 }
 
 func TestTrustDomainID(t *testing.T) {
-	id := spiffeid.Must("domain.test")
+	assert.Zero(t, (spiffeid.TrustDomain{}).ID())
 
-	// Common case
-	td := spiffeid.RequireTrustDomainFromString("spiffe://domain.test/path/element")
-	assert.Equal(t, id, td.ID())
-
-	// Empty path
-	td = spiffeid.RequireTrustDomainFromString("domain.test")
-	assert.Equal(t, id, td.ID())
-}
+	expected := spiffeid.RequireFromString("spiffe://trustdomain")
 
-func TestTrustDomainIDString(t *testing.T) {
-	// Common case
-	td := spiffeid.RequireTrustDomainFromString("spiffe://domain.test/path/element")
-	assert.Equal(t, "spiffe://domain.test", td.IDString())
+	td = spiffeid.RequireTrustDomainFromString("trustdomain")
```

Comment:
nit: could condense the logic with a for loop since every case is asserting the same thing
```go
testCases := []string{"trustdomain", "spiffe://trustdomain", "spiffe://trustdomain/path"}
for _, tdStr := range testCases {
    td = spiffeid.RequireTrustDomainFromString(tdStr)
    assert.Equal(t, expected, td.ID())
}
```

Reply:
👍